plugin/evm/vm.go

```diff
@@ -1218,15 +1226,41 @@ func (vm *VM) setAppRequestHandlers() {
 			},
 		},
 	)
+	if err := vm.RegisterLeafRequestHandler(message.StateTrieNode, "sync_state_trie_leaves", evmTrieDB, message.StateTrieKeyLength, true); err != nil {
+		return err
```

Comment:
nit

```suggestion
		return fmt.Errorf("failed to register state trie database: %s", err)
```

maybe `%w` is needed though